core/app/models/spree/payment.rb

```diff
@@ -97,6 +97,7 @@ def transaction_id
       event :void do
         transition from: [:pending, :processing, :completed, :checkout], to: :void
       end
+      after_transition to: :void, do: :after_void
```

Comment:
@sebastian-palma couldn't we just plug into the `after_transition` block on line 105?

Reply:
Didn't know about that @damianlegawiec, I'm going to try it. Will it run for every transition? Should I somehow check the current event?

Reply:
> @sebastian-palma couldn't we just plug into the `after_transition` block on line 105?

This can't be done right in the core project because `Spree::Webhooks::Endpoints::QueueRequests` is in the api project. We could somehow extend the `Spree::Payment.state_machine` within api, but it'll be only for Spree::Payment @damianlegawiec.